src/app/Main/Body/QueryPage/QueryPage-LoadingOrError.component.tsx

```diff
@@ -21,23 +21,15 @@ export function QueryPageLoadingOrErrorComponent(props: {
     if (props.status !== undefined) {
         if (props.status.isolateStatus !== 200) {
             return (
-                <ErrorPageComponent
-                    errorStatus={props.status.isolateStatus}
-                />
+                <ErrorPageComponent errorStatus={props.status.isolateStatus} />
             );
         }
         if (props.status.filterStatus !== 200) {
             return (
-                <ErrorPageComponent
-                    errorStatus={props.status.filterStatus}
-                />
+                <ErrorPageComponent errorStatus={props.status.filterStatus} />

```

Comment:
General idea for a later refactoring:
I would suggest to use some function: 
fetchQueryPageData(): Status {
status = fetchData1()
if status === 404 return Status.PageNotFound;
if status !== 200 return Status.GeneralError;
status = fetchData2()
if status === 404 ...
if status === 200 ...
return Status.Success
}
And in the QueryPage container compoment:
let status = Status.Loading;
status = fetchQueryPageData(); // this is in some async hook
if(loading) return LoadingComponent
if(status === Status.Success) return QueryPage
else return ErrorPage(status)
Idea is that there is only one status independent of how many request are needed for the fetching all the data. So you don't have to differentiate between isolateStatus and filterStatus. And the actually http status code handling is separated from the rest of your code and at the same place where the fetching happens.